cargo-dist/templates/ci/github_ci.yml.j2

```diff
@@ -229,6 +229,13 @@ jobs:
 
 {{%- endif %}}
 
+{{%- for job in user_publish_jobs %}}
+  user-publish-jobs:
+    needs: [plan, should-publish]
+    if: ${{ !fromJson(needs.plan.outputs.val).announcement_is_prerelease || fromJson(needs.plan.outputs.val).publish_prereleases }}
+    uses: {{{ job }}}
```

Comment:
Oh wait, you want to invoke them? I'm not sure if that will be pleasant (passing stuff between workflow files is a huge pain). I was assuming cargo-dist would load the file and paste its contents into here. Maybe I'm in the wrong though.

Reply:
hmm, i suppose if we just hand them `needs.plan.outputs.val` that's all the input they really need..?

Reply:
Yeah, that gives an input we've already committed to keeping stable(ish) so I think it's a reasonable approach.

Reply:
worth a try

Reply:
This seems happy; I did some tests with this and it's working fine. I added secret inheritance as well, ensuring that user-defined workflows can make use of any secrets in the repository.